alacritty/res/text.f.glsl

```diff
@@ -18,7 +18,7 @@ void main() {
         }
 
         alphaMask = vec4(1.0);
-        color = vec4(bg.rgb, 1.0);
+        color = vec4(bg.rgb, 1.0) * bg.a;
```

Comment:
I'm not sure this logic is correct? Just looking at the results, it seems like `color = vec4(bg.rgb, bg.a);` gives me results closer to what I would expect?

Neither of them seem to actually be correct though, since it seems like they aren't independent from the actual background color? I'd expect a transparent cell background with the same hex color as the background to have the exact same color, but it doesn't. This is likely because the cell's background is getting merged with the background of the terminal.

Reply:
Hm. I vaguely remember thinking about this and preferring it to darken the colours as well when applying the opacity. I'll go through this again and see if I can retrace my steps.

Reply:
As I've stated, I think both are incorrect. Generally the ideal solution would be to do this *correctly*, but I haven't looked into that so I can't show you a solution for that.

But at least `vec4(bg.rgb, bg.a)` looks like it is a little more correct at least on a dark background.

Reply:
Okay, I think the reason I went with the multiplier on all instead of just setting alpha was that the latter messed up the background more visibly when entering tmux (it's a bit difficult to tell, but the issue is the more whiteish background in tmux compared to the non-tmux in the last two screenshots below - the opacity is set to 0.4 to have a more extreme impact, I normally run with 0.9).

I think you are right that there is something fishy going on, it does look like that tmux/no-tmux does look slightly different in my implementation as well. But other than disecting the complete rendering path I don't know how to do it properly - so certainly won't come up with a solution today and possibly not this year either. I'll poke around as I get time over the next days. I *think* that my original implementation (which just backed out changes that switch the original opacity-to-all to the now-in-master opacity-only-to-default-bg) didn't have this problem, so maybe that's a place to start. 

master:
<img width="411" alt="master" src="https://user-images.githubusercontent.com/3343670/103082482-1dc79580-45d2-11eb-83c0-6c0cc4ccb61f.png">
master tmux:
<img width="374" alt="master tmux" src="https://user-images.githubusercontent.com/3343670/103082491-21f3b300-45d2-11eb-9ffb-4366c96d34d9.png">
this branch:
<img width="363" alt="branch" src="https://user-images.githubusercontent.com/3343670/103082492-21f3b300-45d2-11eb-9b5e-06bc01eedfcd.png">
this branch tmux:
<img width="359" alt="branch-tmux" src="https://user-images.githubusercontent.com/3343670/103082493-228c4980-45d2-11eb-86bc-da9b53584308.png">
this branch only alpha:
<img width="366" alt="only-alpha" src="https://user-images.githubusercontent.com/3343670/103082495-23bd7680-45d2-11eb-9efe-d6fbc43bd850.png">
this branch only alpha tmux:
<img width="359" alt="only-alpha-tmux" src="https://user-images.githubusercontent.com/3343670/103082497-24560d00-45d2-11eb-824f-99ee695b0dcb.png">

Reply:
Ah, the difference in bg between tmux/non-tmux case is because of how the background opacity is currently done in master (https://github.com/alacritty/alacritty/blob/master/alacritty/src/renderer/mod.rs#L761-L766), it doesn't just set the alpha channel, it also scales the RGB channels. I don't have an opinion on which is preferable, but changing both `clear` to only set the alpha channel and `color = vec4(bg.rgb, bg.a)` I don't see a difference in bg colour between my tmux/non-tmux cases.
So if you are happy to change how the opacity is applied also in the 'only for the default bg' case, then it works to just apply it to the alpha channel.
For how I am using the opacity setting it doesn't make much of a difference. But for a smaller opacity setting the difference is much more notable, though I find it hard to tell which version 'looks better'.

Reply:
I have no problem with changing things as long as it works properly. There shouldn't be much "opinion" involved. 50% opacity should be a pretty clear instruction usually.

Reply:
¯\_(ツ)_/¯
Changed to only apply the opacity to the alpha channel.

Reply:
You just need to watch out, because the glyph data is not in RGB format.